src/DropdownItem.js

```diff
@@ -40,7 +43,9 @@ class DropdownItem extends React.Component {
       this.props.onClick(e);
     }
 
-    this.context.toggle();
+    if (this.props.toggle) {
+      this.context.toggle();
```

Comment:
I completely agree, I'll do this and create a new PR as soon as I have some time.